src/components/ContextBar.test.js

```diff
@@ -0,0 +1,626 @@
+    machineType: 'Standard_DS1_v2',
+    persistentDiskId: 15778,
+    region: 'eastus',
+    runtimeName: 'saturn-b2eecc2d-75d5-44f5-8eb2-5147db41874a',
+    status: 'Running',
+    workspaceId: '2a77dcb5-882c-46b9-a3bc-5d251aff14d0'
+  }
+}
+
+const runtimeDisk = {
+  id: 15778,
+  googleProject: 'terra-dev-cf677740',
+  cloudContext: {
+    cloudProvider: 'GCP',
+    cloudResource: 'terra-dev-cf677740'
+  },
+  zone: 'us-central1-a',
+  name: 'saturn-pd-c4aea6ef-5618-47d3-b674-5d456c9dcf4f',
+  status: 'Ready',
+  auditInfo: {
+    creator: 'testuser123@broad.com',
+    createdDate: '2022-07-18T18:35:32.012698Z',
+    destroyedDate: null,
+    dateAccessed: '2022-07-18T20:34:56.092Z'
+  },
+  size: 50,
+  diskType: {
+    label: 'pd-standard',
+    displayName: 'Standard',
+    regionToPricesName: 'monthlyStandardDiskPrice'
+  },
+  blockSize: 4096,
+  labels: {
+    saturnWorkspaceNamespace: 'general-dev-billing-account',
+    saturnWorkspaceName: 'Broad Test Workspace'
+  }
+}
+
+const rstudioRuntime = {
+  id: 76979,
+  workspaceId: null,
+  runtimeName: 'saturn-48afb74a-15b1-4aad-8b23-d039cf8253fb',
+  googleProject: 'terra-dev-98897219',
+  cloudContext: {
+    cloudProvider: 'GCP',
+    cloudResource: 'terra-dev-98897219'
+  },
+  auditInfo: {
+    creator: 'ncl.hedwig@gmail.com',
+    createdDate: '2022-09-08T19:46:37.396597Z',
+    destroyedDate: null,
+    dateAccessed: '2022-09-08T19:47:21.206Z'
+  },
+  runtimeConfig: {
+    machineType: 'n1-standard-4',
+    persistentDiskId: 15778,
+    cloudService: 'GCE',
+    bootDiskSize: 120,
+    zone: 'us-central1-a',
+    gpuConfig: null
+  },
+  proxyUrl: 'https://leonardo.dsde-dev.broadinstitute.org/proxy/terra-dev-98897219/saturn-48afb74a-15b1-4aad-8b23-d039cf8253fb/rstudio',
+  status: 'Running',
+  labels: {
+    saturnWorkspaceNamespace: 'general-dev-billing-account',
+    'saturn-iframe-extension': 'https://bvdp-saturn-dev.appspot.com/jupyter-iframe-extension.js',
+    creator: 'ncl.hedwig@gmail.com',
+    clusterServiceAccount: 'pet-26534176105071279add1@terra-dev-98897219.iam.gserviceaccount.com',
+    saturnAutoCreated: 'true',
+    clusterName: 'saturn-48afb74a-15b1-4aad-8b23-d039cf8253fb',
+    saturnWorkspaceName: 'N8s Space',
+    saturnVersion: '6',
+    tool: 'RStudio',
+    runtimeName: 'saturn-48afb74a-15b1-4aad-8b23-d039cf8253fb',
+    cloudContext: 'Gcp/terra-dev-98897219',
+    googleProject: 'terra-dev-98897219'
+  },
+  patchInProgress: false
+}
+
+const contextBarProps = {
+  runtimes: [],
+  apps: [],
+  appDataDisks: [],
+  refreshRuntimes: () => '',
+  location: 'US-CENTRAL1',
+  locationType: '',
+  refreshApps: () => '',
+  workspace: {
+    workspace: {
+      namespace: 'namespace'
+    },
+    namespace: 'Broad Test Workspace'
+  }
+}
+
+describe('ContextBar - buttons', () => {
+  it('will render default icons', () => {
+    // Act
+    render(h(ContextBar, contextBarProps))
+
+    // Assert
+    expect(screen.getByText('Rate:'))
```

Comment:
Instead of querying all of the document via `screen`, can these tests use the queries returned by `render`?

https://testing-library.com/docs/react-testing-library/api#queries-1

Reply:
Is there a particular advantage or difference to this? I didn't realize it was an option!

Reply:
I'll claim it's more idiomatic.

It may also be useful if there's a case like a component opens a modal and you want to query the component or the modal specifically.

Reply:
Ahh I see now. Let me make the changes!

Reply:
Also good for test isolation. If another test leaves some elements in the DOM, querying with screen could pick those up.